packages/calypso-e2e/src/lib/utils/validate-translations.ts

```diff
@@ -0,0 +1,121 @@
+import { sprintf } from '@wordpress/i18n';
+import { Page, ElementHandle } from 'playwright';
+
+const GLOTPRESS_ORIGINALS_ENDPOINT =
+	'https://translate.wordpress.com/api/translations/-query-by-originals';
+const GLOTPRESS_PROJECT = 'wpcom';
+
+const selectors = {
+	translatableElement: '[data-e2e-string]',
+};
+
+type OriginalString = { singular: string; plural?: string; context?: string };
+type Translation = {
+	original: OriginalString;
+	translations: {
+		translation_0: string;
+	}[];
+};
+
+/**
+ * Fetch translations for originals.
+ */
+async function fetchTranslations(
+	page: Page,
+	originals: OriginalString[],
+	locale: string
+): Promise< Translation[] | null > {
+	const params = {
+		url: GLOTPRESS_ORIGINALS_ENDPOINT,
+		data: {
+			project: GLOTPRESS_PROJECT,
+			locale_slug: locale,
+			original_strings: JSON.stringify( originals ),
+		},
+	};
+
+	return page.evaluate( async ( { url, data } ) => {
+		const body = new URLSearchParams();
+		for ( const key in data ) {
+			body.append( key, data[ key as 'project' | 'locale_slug' | 'original_strings' ] );
+		}
+
+		return window
```

Comment:
There is no specific reason to send the request from the page. I just wanted to use `fetch` as it has cleaner syntax and sending it through the page was the workaround I came up with not to introduce new dependencies for a single request.

However, I see why this could be confusing or even problematic in some cases, so I've switched it to using `request` from the core `https` lib - https://github.com/Automattic/wp-calypso/pull/57707/commits/4672e76befb6a1f561b7b8ca8ba7ff8849101301#diff-c9f4039cb8be26088b4672860ea55820f2f13bbc08555f87923c9b181c2a43dbR41.

Reply:
Oh man, no kidding - that original `https` library is pretty primitive!

What you have looks great, but if you want that cleaner `fetch` syntax, it looks like the root `package.json` has `node-fetch`, and you can totally add that to this packages `package.json`!

By no means required, what you have works great, just an option if you want that better syntax :)

Reply:
Ha, I must have overlooked the root dependencies, because I didn't see it when I checked. Looks much cleaner with `node-fetch` - https://github.com/Automattic/wp-calypso/pull/57707/commits/08bb262ed7891294a9f064f42d83e3f9f94a1b31#diff-c9f4039cb8be26088b4672860ea55820f2f13bbc08555f87923c9b181c2a43dbR45-R56.

Reply:
Wow!!  That really _is_ such a cleaner API, that looks so nice! Looks great, thanks for the update! 😄  👍